jscomp/runtime/caml_obj.ml

```diff
@@ -167,6 +167,8 @@ let unsafe_js_compare x y =
 let rec caml_compare (a : Obj.t) (b : Obj.t) : int =
   if a == b then 0 else
   (*front and formoest, we do not compare function values*)
+  if a == (Obj.repr Js_null.empty) then -1 else
+  if b == (Obj.repr Js_null.empty) then 1 else
```

Comment:
The `a = null` is caught,  would not the  case when `b = null` be moved to line 192 ?

Reply:
That's where I had it before. I've moved it here because caml_int_compare and  caml_string_compare give inconsistent results when comparing with null.

Reply:
E.g. `caml_int_compare(-1,null) == -1` but `caml_int_compare(1,null) == 1`.

Reply:
I am a little confused why it is weird? `-1 < null < 1`?